### PR TITLE
fix: only initialize gleap after developer survey and cookie promp

### LINKED ISSUE #1277: Refactor Cookie Prompt
The cookie prompt does not initiate the developer survey when interacted with. This also means that the Gleap widget remains hidden. To provide a better UX, one which does not require the user to refresh or visit another page, a refactor of the cookie prompt's hook to a store would be a way to ensure that once dismissed the developer survey and Gleap button will display immediately.

conversation and pr can be found [here](https://github.com/near/near-discovery/pull/1272)

### PR DESCRIPTION
**Pull request by jackson-harris-iii**
This ticket hides the gleap widget until a user has completed interactions with the developer survey and cookie prompt ensuring there are no UI/UX issues between the overlays.

Addressing a concern raised in this [issue](https://github.com/near/near-discovery/issues/1258)

Closes #1277

### CODE REVIEW

**Comment by calebjacob** [comment]
Hey @jackson-harris-iii, I'm seeing some issues just testing the preview build:

- Once I accept cookies, the developer survey doesn't pop up right away. I have to reload my browser for it to show up.
- If I click the `X` button to dismiss the survey, the Gleap icon never shows up. If I reload my browser, it still doesn't show up either.

**Review by calebjacob — CHANGES_REQUESTED** [review]
See comment above. Thanks!

**Comment by jackson-harris-iii** [comment]
> Hey @jackson-harris-iii, I'm seeing some issues just testing the preview build:
> 
> * Once I accept cookies, the developer survey doesn't pop up right away. I have to reload my browser for it to show up.
> * If I click the `X` button to dismiss the survey, the Gleap icon never shows up. If I reload my browser, it still doesn't show up either.

The issue was caused by depending on hooks for the form interactions instead of listening to a shared state (like in a store.) When we instantiate a hook, like the pattern used for the cookie prompt and the previous version of the form, a new instance is created every time. Hoisting the variable being watched to a shared store has resolved the bug.

**Comment by calebjacob** [comment]
@jackson-harris-iii Just tested out the preview build again. I'm still seeing this issue:

- Once I accept cookies, the developer survey doesn't pop up right away. I have to reload my browser for it to show up.

This issue is fixed though:

- If I click the X button to dismiss the survey, the Gleap icon never shows up. If I reload my browser, it still doesn't show up either.

**Comment by jackson-harris-iii** [comment]
@calebjacob the cookies accept/decline uses hooks and doesn't have a store. We could refactor this to use a store instead so that the survey is triggered immediately or the user will be prompted with the survey at the next page change.

**Comment by calebjacob** [comment]
@jackson-harris-iii Sure, feel free to refactor the cookie prompt to use a store. I think it'll be a better UX for the survey prompt to pop up right when you accept cookies. From what I've seen and heard, the Gleap help widget actually gets used quite a bit by our users and we should make sure it's still easy to access after accepting cookies and dismissing or filling out the survey. 👍

**Comment by jackson-harris-iii** [comment]
@calebjacob @shelegdmitriy , the cookie prompt is a BOS component and there is a listener in the VM that handles updating the app and localstorage. This is the first I've seen a BOS component interact with the vercel app. So I wanted to mention this in the conversation for visibility. 

https://github.com/near/near-discovery/blob/5f6e665ba92fc2efe3939c8ea9ad4bcef6ef8211/src/components/vm/VmInitializer.tsx#L167

**Comment by charleslavon** [comment]
@jackson-harris-iii An alternate to using a zustand store, would it be more simple to add/remove listener's which inspect the [storage events](https://developer.mozilla.org/en-US/docs/Web/API/Window/storage_event) such that when can respond to the trigger of saving into local storage the survey acknowledged/closed data.

**Review by jackson-harris-iii — COMMENTED** [review]
Necessary changes have been made to ensure that the dismissal of the cookie prompt will trigger a render of the developer survey.

File: src/stores/cookieData.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import { create } from 'zustand';
+
+export type CookieDataState = {
+  cookieData: boolean;
+  set: (state: Partial<CookieDataState>) => void;
+  checkCookieData: () => void;
+};
+
+type CookieDataStore = CookieDataState & {
+  set: (state: Partial<CookieDataState>) => void;
```
Comment (charleslavon) [inline src/stores/cookieData.ts:10]: @jackson-harris-iii why is it necessary to define `set` and `checkCookieData` in both the State and Store types?

**Comment by charleslavon** [comment]
@jackson-harris-iii LGTM! one minor question

File: src/stores/cookieData.ts (hunk 1 above)
Comment (jackson-harris-iii) [reply src/stores/cookieData.ts:10]: @charleslavon I was able to remove it from the state types.

**Review by charleslavon — APPROVED** [review]

**Pull request merged**